.github/workflows/gradle-run-tests.yml

```diff
@@ -0,0 +1,35 @@
+name: "Gradle Run Tests"
+
+on:
+  pull_request:
+    types: [ opened, synchronize ] # Don't rerun on `edited` to save time
+
+jobs:
+  gradle-run-tests:
+    runs-on: macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          cache: gradle
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+      - name: Test with Gradle
+        run: ./gradlew test :network:allTests
```

Comment:
Is the idea that when we add more modules we add something like this here:
```
 - name: Test with Gradle (Cryptography)
  run: ./gradlew test :cryptograhy:allTests
```

or will it just become

```
  run: ./gradlew test
```

when the project is clean and all modules have working tests?

Reply:
I discovered there was no problem just running directly all tests. So I changed it to `./gradlew test` 

To get tests results we still need to add some lines as we expand the tests.